src/main/groovy/nebula/plugin/publishing/PublicationBase.groovy

```diff
@@ -0,0 +1,87 @@
+package nebula.plugin.publishing
+
+import nebula.plugin.publishing.ivy.IvyBasePublishPlugin
+import nebula.plugin.publishing.maven.MavenBasePublishPlugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.ivy.IvyPublication
+import org.gradle.api.publish.ivy.plugins.IvyPublishPlugin
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+
+trait PublicationBase implements Plugin<Project>, StandardTextValues {
+
+    /**
+     * Checks to see if the task already exists or not on the project.  If it does not, then it creates it.
+     * @param project project to add the task too
+     * @param task Map configuration of the task to create
+     * @return Null if it was not created, the task object if it was.
+     */
+    Task addTaskLocal(Project project, Map<String, ?> task) {
+        if (!project.tasks.findByName(task.get('name').toString())) {
+            project.tasks.create(task)
+        } else {
+            null
+        }
+    }
+
+    void checkAddMavenPublish(Project project, Task createdTask) {
+        project.plugins.withType(MavenPublishPlugin) {
+            project.plugins.apply(MavenBasePublishPlugin)
+
+            project.publishing.publications {
```

Comment:
http://stackoverflow.com/questions/21190230/how-can-plugin-programmatically-configure-maven-publish-publishing-and-allow-bui

Apparently Justin Ryan built a class in a very early version of nebula publish to try to handle this specific issue.  but just changing the syntax is far easier than his fix.  

It looks like i have no choice but to build an additional custom plugin (a build.gradle file is nothing more than a plugin).  It seams you can't replicate the problem from a single plugin closure, like a build.gradle file.  It seems like you can only replicate it from a parent/child plugin relationship, where the parent introduces publishing and finishes its apply method, then a child method attempts to access the publishing from its apply method.  At least i think so.